endpoint.go

```diff
@@ -767,13 +767,15 @@ func (ep *endpoint) assignAddress(assignIPv4, assignIPv6 bool) error {
 	}
 
 	if assignIPv4 {
-		if err = ep.assignAddressVersion(4, ipam); err != nil {
+		if err = ep.assignAddressVersion(4, ipam); err != nil && !ep.network.enableIPv6 {
 			return err
 		}
 	}
 
 	if assignIPv6 {
-		err = ep.assignAddressVersion(6, ipam)
+		if err = ep.assignAddressVersion(6, ipam); err != nil && ep.network.enableIPv6 {
```

Comment:
I just noticed that this change is not needed at all because the error is returned a couple of lines down anyway. I will drop this.